packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -136,18 +145,31 @@ contract SignalService is AuthorizableContract, ISignalService {
             Hop memory hop = p.hops[i];
             if (hop.stateRoot == stateRoot) revert SS_INVALID_HOP_PROOF();
 
-            bytes32 label = authorizedAddresses[hop.relayerContract];
+            bytes32 label = authorizedAddresses[hop.relay];
             if (label == 0) revert SS_HOP_RELAYER_UNAUTHORIZED();
 
             uint64 hopChainId = uint256(label).toUint64();
 
+            // Do not cache hop, just the source chain's signal service storage root.
```

Comment:
Would cache the hops as well: https://github.com/taikoxyz/taiko-mono/pull/15683#discussion_r1484217054